Sources/GDBRemote/Session.cpp

```diff
@@ -3096,6 +3096,17 @@ void Session::Handle_vFile(ProtocolInterpreter::Handler const &,
     } else {
       ss << 'F' << std::hex << size;
     }
+  } else if (op == "mode") {
+    uint32_t mode;
+    ErrorCode error =
+        _delegate->onFileGetMode(*this, HexToString(&args[op_end]), mode);
+    // Response is F followed by the mode bits in base 16 or
+    // F-1,errno with the errno if an error occurs, base 16
+    if (error != kSuccess) {
+      ss << 'F' << -1 << ',' << std::hex << error;
+    } else {
+      ss << 'F' << std::hex << mode;
+    }
```

Comment:
Why not flip this around?

```suggestion
    ss << 'F';
    if (error == kSuccess)
      ss << -1 << ',' << std::hex << error;
    else
      ss << std::hex << mode;
```

Reply:
All of the other cases in this function order the error check this way. I would argue to leave it the way it is for consistency.

Reply:
Okay, however, I think that we should extract the packet construction to not replicate the package prefix (`F`).